clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/DisableAppEngineAtomicOperation.groovy

```diff
@@ -0,0 +1,114 @@
+
+  @Autowired
+  AppEngineLoadBalancerProvider appEngineLoadBalancerProvider
+
+  @Autowired
+  AppEngineClusterProvider appEngineClusterProvider
+
+  DisableAppEngineAtomicOperation(EnableDisableAppEngineDescription description) {
+    this.description = description
+  }
+
+  @Override
+  Void operate(List priorOutputs) {
+    task.updateStatus BASE_PHASE, "Initializing disable server group operation for $description.serverGroupName in $description.credentials.region..."
+
+    def credentials = description.credentials
+    def serverGroupName = description.serverGroupName
+
+    task.updateStatus BASE_PHASE, "Looking up server group $serverGroupName..."
+    def serverGroup = appEngineClusterProvider.getServerGroup(credentials.name, credentials.region, serverGroupName)
+    def loadBalancerName = serverGroup?.loadBalancers?.first()
+
+    def loadBalancer = appEngineLoadBalancerProvider.getLoadBalancer(credentials.name, loadBalancerName)
+
+    if (!loadBalancer.split.allocations.containsKey(serverGroupName)) {
+      task.updateStatus BASE_PHASE, "Server group $serverGroupName does not receive traffic from load balancer $loadBalancer.name," +
+        " ending operation..."
+      return null
+    }
+
+    def split = buildTrafficSplitWithoutServerGroup(loadBalancer.split, serverGroupName)
+
+    def upsertLoadBalancerDescription = new UpsertAppEngineLoadBalancerDescription(
+      credentials: credentials,
+      loadBalancerName: loadBalancerName,
+      split: split,
+      migrateTraffic: description.migrateTraffic
+    )
+
+    task.updateStatus BASE_PHASE, "Updating load balancer $loadBalancerName..."
+    def upsertLoadBalancerOperation = new UpsertAppEngineLoadBalancerAtomicOperation(upsertLoadBalancerDescription)
+    upsertLoadBalancerOperation.appEngineLoadBalancerProvider = appEngineLoadBalancerProvider
+    upsertLoadBalancerOperation.operate(priorOutputs)
+
+    return null
+  }
+
+  // https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps.services#TrafficSplit
+  private static final COOKIE_SPLIT_PRECISION = 1000
+  private static final IP_SPLIT_PRECISION = 100
+
+  static AppEngineTrafficSplit buildTrafficSplitWithoutServerGroup(AppEngineTrafficSplit oldSplit, String serverGroupName) {
+    AppEngineTrafficSplit newSplit = oldSplit.clone()
+
+    // The validator ensured that the server group we're disabling doesn't have an allocation of 1, which would be bad.
+    Double denominator = 1 - oldSplit.allocations.get(serverGroupName)
+    newSplit.allocations.remove(serverGroupName)
+
+    def precision = newSplit.shardBy == ShardBy.COOKIE ? COOKIE_SPLIT_PRECISION : IP_SPLIT_PRECISION
+
+    newSplit.allocations = newSplit.allocations.collectEntries { name, allocation ->
+      Double newAllocation = Math.round(allocation / denominator * precision) / precision
+
+      [(name): newAllocation]
+    } as Map<String, Double>
+
+    Double error = 1 - newSplit.allocations.collect { k, v -> v }.sum()
```

Comment:
`allocation / denominator * precision` might not behave like you expect it to due to floating point error. It could be that your sum of allocations winds up being greater than 1 (I think), and if the newest server group has an allocation of 0, now you're in trouble.